tests/internal/test_safety.py

```diff
@@ -57,3 +62,21 @@ def test_safe_function_call():
 def test_safe_eval_property():
     with pytest.raises(AttributeError):
         SafeObjectProxy.safe(UnsafeObject()).unsafe_property
+
+
+@pytest.mark.parametrize("_type", [list, set, tuple, frozenset])
+def test_safe_collection(_type):
```

Comment:
docstrings would be nice, with minimal head scratching I can figure out what this is testing, but a very simple sentence would definitely help